packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -1255,6 +1280,8 @@ Hardhat Network's forking functionality only works with blocks from at least spu
             );
           }
 
+          vm._common.setHardfork(this._selectHardfork(new BN(blockNumber)));
```

Comment:
This line surprises me a lot. Shouldn't we directly construct the new common (i.e. in `this._getCommonForTracing`) with the right hf?

Reply:
upadte: it already does

Reply:
I think this is leftover code (:grimacing:) from when we weren't instantiating a new common. Addressed in https://github.com/nomiclabs/hardhat/pull/1928/commits/3f50f0d620206c19642a9c82eecd8847073a6822